File: 0189-rotate-array/0189-rotate-array.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var rotate = function(nums, k) {
      k = k % nums.length;
   
  const a = nums.splice(-k); 
  nums.unshift(...a);
  return nums;
    


    
};

File: 0189-rotate-array/0189-rotate-array.java
class Solution {
    public void rotate(int[] nums, int k) {

        int[] a=new int[nums.length];
        int n=nums.length;
        k=k%n;
        

        for(int i=0;i<nums.length;i++){
            int newi=(i+k)%n;
            a[newi]=nums[i];

        }
        for(int i=0;i<n;i++){
            nums[i]=a[i];
        }
      
        
    }
}